src/main/scala/com/fulcrumgenomics/util/miseq/Sample.scala

```diff
@@ -71,5 +73,26 @@ class Sample(val sampleOrdinal: Int,
       case Some(str) => Some(str.trim)
     }
   }
+
+  /** Sets the sample barcode based on the read structure.
+    *
+    * @param sampleBarcodeReadStructures a read structure per read (ex. i7 and i5) that contains only sample barcode segments.
+    */
+  def setSampleBarcode(sampleBarcodeReadStructures: Seq[ReadStructure]): this.type = {
```

Comment:
Also it seems like the correlation between read and index is well known, but between read and ReadStructure is inferred based on position in the Seq?  That seems dangerous to me.